v1.18 build failure on Linux & macOS

Hello 👋 . I'm a maintainer for the [Homebrew](https://brew.sh) project. While packaging v1.18 of utftex we are encountering a build failure:

```
  make[1]: Entering directory '/tmp/utftex-20230524-5877-efo77/libtexprintf-1.18/src'
  /bin/bash ./gen_errorflags.sh ./boxes.c ./drawbox.c ./lexer.c ./parser.c
  /bin/bash ./gen_errorflags.sh ./boxes.c ./drawbox.c ./lexer.c ./parser.c
  Collecting error flags from ./boxes.c
  Collecting error flags from ./boxes.c
  Collecting error flags from ./drawbox.c
  Collecting error flags from ./drawbox.c
  Collecting error flags from ./lexer.c
  Collecting error flags from ./lexer.c
  Collecting error flags from ./parser.c
  Collecting error flags from ./parser.c
  rm: cannot remove 'tmperrflags': No such file or directory
  make[1]: *** [Makefile:1079: errorflags.h] Error 1
  make[1]: Leaving directory '/tmp/utftex-20230524-5877-efo77/libtexprintf-1.18/src'
  make: *** [Makefile:674: install-recursive] Error 1
```
The relevant GitHub Actions run can be found [here](https://github.com/Homebrew/homebrew-core/actions/runs/5070774151).

Relates to https://github.com/Homebrew/homebrew-core/pull/131899

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -28,5 +28,10 @@ EXTRA_DIST = utftex.1 utfstringinfo.1 gen_errorflags.sh
 
 BUILT_SOURCES = errorflags.h errormessages.h
 
-errorflags.h errormessages.h: gen_errorflags.sh boxes.c drawbox.c lexer.c parser.c
+errorflags.h: gen_errorflags.sh boxes.c drawbox.c lexer.c parser.c
 	${SHELL} ${srcdir}/gen_errorflags.sh ${srcdir}/boxes.c ${srcdir}/drawbox.c ${srcdir}/lexer.c ${srcdir}/parser.c
+
+# see https://www.gnu.org/software/automake/manual/html_node/Multiple-Outputs.html
+errormessages.h: errorflags.h
+	@test -f $@ || rm -f errorflags.h
+	@test -f $@ || $(MAKE) $(AM_MAKEFLAGS) errorflags.h

